src/crypto.js

```diff
@@ -0,0 +1,274 @@
+const libp2pCrypto = require('libp2p-crypto')
+const TextDecoder = require('text-encoding').TextDecoder
+const TextEncoder = require('text-encoding').TextEncoder
+const { encode, decode } = require('base64-arraybuffer')
+const { Buffer } = require('buffer')
+const multihashing = require('multihashing-async')
+const { pem, pki } = require('node-forge')
+const peerId = require('peer-id')
+
+const { log, error } = require('./log')
+
+const { OBJECT, STRING, UNDEFINED,
+        ARRAY, INTEGER, BOOL } = require('./utils')
+
+const SHA_256 = 'sha2-256'
+
+class Crypto {
+
+  constructor (node) {
+    if (!node) { throw new Error('IPFS node is required') }
+    this._node = node
+  }
+
+  get identity () {
+    if (this._identity) {
+      return this._identity
+    }
+    error('Set identity object required')
+    return null
+  }
+
+  set identity (identity) {
+    if (!identity) {
+      throw new Error('Identity object required')
+    }g
```

Comment:
Even if you didn't want to throw an error, throw one anyway! (`g`)

Reply:
LOL. I blame emacs and `control-g`. Thanks.